docs/storage/volume_migration.md

```diff
@@ -0,0 +1,222 @@
+# Migration update volume strategy and volume migration
+
+Storage migration is possible while the VM is running by using the update volume strategy. Storage migration can be useful in the cases where the users need to change the underlying storage, for example, if the storage class has been deprecated, or there is a new more performant driver available.
+
+This feature doesn't handle the volume creation or cover migration between storage classes, but rather implement a basic API which can be used by overlaying tools to perform more advanced migration planning.
+
+If `Migration` is specified as `updateVolumesStrategy`, KubeVirt will try to migrate the storage from the old volume set to the new one when the VirtualMachine spec is updated. The migration considers the changed volumes present into a single update. A single update may contain modifications to more than one volume, but sequential changes to the volume set will be handled as separate migrations.
+
+The updates are declarative and [GitOps](https://kubevirt.io/user-guide/cluster_admin/gitops/) compatible. For example, a new version of the VM specification with the new volume set and the migration volume update strategy can be directly applied using `kubectl apply` or interactively editing the VM with `kubectl edit`
+
+Example:
+Original VM with a datavolume and datavolume template:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: vm-dv
+  name: vm-dv
+spec:
+  dataVolumeTemplates:
+  - metadata:
+      creationTimestamp: null
+      name: src-dv
+    spec:
+      storage:
+        accessModes:
+        - ReadWriteOnce
+        volumeMode: Filesystem
+        resources:
+          requests:
+            storage: 2Gi
+        storageClassName: local
+      source:
+        registry:
+          url: docker://registry:5000/kubevirt/alpine-container-disk-demo:devel
+  runStrategy: "Always"
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-dv
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: datavolumedisk1
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 128Mi
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - dataVolume:
+          name: src-dv
+        name: datavolumedisk1
+      networks:
+      - name: default
+        pod: {}
+```
+
+The datavolume `src-dv` is migrated to the pvc `dest-pvc`:
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: dest-pvc
+spec:
+  accessModes:
+  - ReadWriteMany
+  volumeMode: Block
+  resources:
+    requests:
+      storage: 5Gi
+```
+
+by updating the VM:
```

Comment:
I got confused when reading this since the elements were sorted differently in this version.

Maybe too technical - I'd want to have emphasized what to change, so I can see it better. Maybe use a diff here to show what exactly I need to add and/or change. Or maybe specify in a block what has been added or changed?

Reply:
My idea, here, was that the user could copy the 2 full yamls as example and applied them separately, but you are probably right a diff will be better